enterprise/core-edge/src/test/java/org/neo4j/coreedge/scenarios/CoreToCoreCopySnapshotIT.java

```diff
@@ -251,4 +253,9 @@ private String string( int numberOfCharacters )
         }
         return s.toString();
     }
+
+    private CoreState getCoreState( CoreGraphDatabase follower )
+    {
+        return follower.getDependencyResolver().resolveDependency( CoreState.class );
```

Comment:
The introduction of the SPI was creating a circular dependency between platform and edition modules for cleaning up code that is basically it is used in test only.

Moreover the EditionSPI was `null` but for the core-edge edition module which is not a nice design.